synapse/http/site.py

```diff
@@ -55,6 +55,7 @@ def __init__(self, channel, *args, **kw):
         self.site = channel.site
         self._channel = channel  # this is used by the tests
         self.authenticated_entity = None
+        self.target_user = None
```

Comment:
should probably have some comments to explain what these mean.

Reply:
can we just replace both with a `requester` property?

Reply:
Hmm, annoyingly this is also used by the federation code which doesn't have a `Requester` object.

Reply:
I made `SynapseRequest.requester` be either a `Requester` or `str`, which felt the cleanest.